### PR TITLE
Query: Implements support for collection nav. prop translation.

### PR DESCRIPTION
**Pull request by anpete**

### CODE REVIEW

File: test/EntityFramework.Sqlite.FunctionalTests/QueryNavigationsInMemoryTest.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+namespace Microsoft.Data.Entity.Sqlite.FunctionalTests
+{
+    public class QueryNavigationsSqliteTest : QueryNavigationsTestBase<NorthwindQuerySqliteFixture>
+    {
+        public QueryNavigationsSqliteTest(NorthwindQuerySqliteFixture fixture)
+            : base(fixture)
+        {
+        }
+    }
+}
```
Comment (natemcmaster) [inline test/EntityFramework.Sqlite.FunctionalTests/QueryNavigationsInMemoryTest.cs]: No assertions that SQLite query-gen is ok?

Comment (anpete) [inline test/EntityFramework.Sqlite.FunctionalTests/QueryNavigationsInMemoryTest.cs]: The queries are running. I can add some SQL verification.

**Comment by anpete** [comment]
@maumar

**Comment by anpete** [comment]
@maumar

File: src/EntityFramework.Relational/Query/RelationalResultOperatorHandler.cs [hunk 2]
```diff
@@ -150,7 +153,7 @@ var predicate
                 return TransformClientExpression<bool>(handlerContext);
             }
 
-            return handlerContext.EvalOnClient;
+            return handlerContext.EvalOnClient();
```
Comment (maumar) [inline src/EntityFramework.Relational/Query/RelationalResultOperatorHandler.cs]: add requiresClientResultOperator: true for readability, as we usually default to false

File: test/EntityFramework.Core.FunctionalTests/QueryNavigationsTestBase.cs [hunk 3]
```diff
@@ -272,6 +267,210 @@ var orders
...
+                var customers
+                    = (from c in context.Set<Customer>()
+                        select new { Any = c.Orders.Any() }).ToList();
+
+                Assert.Equal(91, customers.Count);
+            }
+        }
+
+        [Fact]
+        public virtual void Collection_where_nav_prop_any()
```
Comment (maumar) [inline test/EntityFramework.Core.FunctionalTests/QueryNavigationsTestBase.cs]: add any with predicate?

File: test/EntityFramework.Core.FunctionalTests/QueryNavigationsTestBase.cs [hunk 4]
```diff
@@ -272,6 +267,210 @@ var orders
...
+        [Fact]
+        public virtual void Collection_select_nav_prop_any()
+        {
+            using (var context = CreateContext())
+            {
+                var customers
+                    = (from c in context.Set<Customer>()
+                        select new { Any = c.Orders.Any() }).ToList();
+
+                Assert.Equal(91, customers.Count);
```
Comment (maumar) [inline test/EntityFramework.Core.FunctionalTests/QueryNavigationsTestBase.cs]: might be good idea to add some more verification, those queries will always return 91 results as there is no predicate

Comment (anpete) [inline test/EntityFramework.Core.FunctionalTests/QueryNavigationsTestBase.cs]: Will do.

File: test/EntityFramework.Core.FunctionalTests/QueryNavigationsTestBase.cs (hunk 3 above)
Comment (anpete) [inline test/EntityFramework.Core.FunctionalTests/QueryNavigationsTestBase.cs]: Will do.

File: src/EntityFramework.Relational/Query/RelationalResultOperatorHandler.cs (hunk 2 above)
Comment (anpete) [inline src/EntityFramework.Relational/Query/RelationalResultOperatorHandler.cs]: Sure.

**Comment by maumar** [comment]
:shipit:  otherwise

File: src/EntityFramework.Relational/Query/RelationalResultOperatorHandler.cs (hunk 2 above)
Comment (anpete) [inline src/EntityFramework.Relational/Query/RelationalResultOperatorHandler.cs]: Going to leave this as the default is true.

**Pull request merged**